08 user login

Claudia108 opened the pull request:
Updates the following features:
1. Add README
2. Fix user login path (the original redirect_to request.referrer interfered with google-oauth and broke the site):
ApplicationController: create before_filter and after_sign_in_path(current_user).
SessionsController: use redirect_to  after_sign_in_path(current_user).
1. Remove delete feature for sites - per client request
2. Admin/SitesController: remove destroy action
3. views/sites/show: remove delete button
4. routes: admin/sites: remove destroy action
5. test/admin_site_creation_test.rb: update last test to mirror that admin cannot delete sites anymore.
6. Add attributes to sites - per client request
7. Add migration to add columns :trinominal and :description and to add attachment :image - as optional attributes.
8. Add migration to add default values to :trinominal and :description if non are given.
9. Update views/admin/sites/_site_form.html.erb and views/sites/show.html.erb with added attributes.
10. Update admin_site_creation_test.rb and user_views_sites_test.rb to reflect changes in schema.
11. Update map.js with mouseover function
12. Add mouseover eventListener to map.js so users can view the name of the site before clicking on it.

Claudia108 committed:
Add README

Claudia108 committed:
Remove delete feature from admin sites

Add before_filter and update redirect path in sessions_controller

Claudia108 committed:
Add description, trinominal and image to sites

Update site views and site tests

marinacor1 commented on app/controllers/application_controller.rb:
```diff
@@ -19,4 +21,18 @@ def admin_authenticated
   end
+
+  def store_location
+    return unless request.get?
```
I haven't seen a return unless like this in Ruby before. What exactly is it doing?

marinacor1 replied:
I know you haven't pushed to Heroku but wanted to add some notes on my experience as a user:

I clicked on the Login using Google link and was able to put in my password and go through the OAuth experience. However, it redirected me to the standard Heroku 404 page.  When I navigated back to the landing page I noticed that the OAuth had worked because my google name was visible. Thought that was interesting!

AllPurposeName commented on app/controllers/application_controller.rb:
```diff
@@ -19,4 +21,18 @@ def admin_authenticated
+
+  def after_sign_in_path(current_user)
+    session[:previous_url] || root_path
+  end
```
Why is `current_user` a parameter here? I don't see it being used.

AllPurposeName commented on app/controllers/application_controller.rb:
```diff
@@ -19,4 +21,18 @@ def admin_authenticated
+    if (request.path != "/auth/google_oauth2" &&
+        request.path != "/auth/google_oauth2/callback" &&
+        request.path != "/logout" &&
+        !request.xhr?) # don't store ajax calls
```
Consider breaking this out into a private method. Something like `if storable?(request)`. All of the conditions are confusing and the oauth strings seem arbitrary or magical.

Claudia108 committed:
Simplify before_action for user login and add tests

Claudia108 committed:
Update path helper-method and move to SessionsController

Claudia108 commented:
Thank you to feedback I simplified the before_action in the ApplicationController and instead added skip_before_action in SessionsController and Api/V1/ApiController.
I also removed the argument from path helper-method in ApplicationController. Was not needed :)

Claudia108 commented:
@AllPurposeName Thank you for your feedback. I simplified the before_filter in ApplicationController to before_action and added skip_before_action in SessionsController and Api/V1/ApiController.
I also removed the argument from the path helper-method and moved this method to the SessionsController(where it is used).
Merging to master in order to move forward with next features.

AllPurposeName commented:
Great solutions! 👍

Claudia108 commented:
@AllPurposeName Thank you!
